server/src/webserver.rs

```diff
@@ -645,18 +656,30 @@ async fn serve_file(
         source_lexer(&file_contents, lexer)
     };
 
-    // <p>Convert doc blocks from Markdown to HTML</p>
+	// <p>Convert doc blocks from Markdown to HTML then parse using HTML5Ever</p>
     let mut options = Options::empty();
     options.insert(Options::ENABLE_STRIKETHROUGH);
     for code_doc_block in &mut code_doc_block_arr {
         if let CodeDocBlock::DocBlock(ref mut doc_block) = code_doc_block {
             let parser = Parser::new_ext(&doc_block.contents, options);
             let mut html_output = String::new();
             html::push_html(&mut html_output, parser);
-            doc_block.contents = html_output;
+		
+		// <p>Create a new RcDom object to represent the DOM tree with default settings</p>
+        let dom = RcDom::default();
+		// <p>Convert html_output to UTF-8 encoding then use parse_fragment to parse the HTML with default options</p>
+        let parse_result = html5ever::parse_fragment(dom,ParseOpts::default(),QualName::new(None, ns!(), "html".into()),Default::default(),).from_utf8().read_from(&mut html_output.as_bytes()).unwrap();
+		// <p>Create an empty vector for html_parsed_output</p>
+        let mut html_parsed_output = Vec::new();
+		// <p>Take the parsed html result document field, clone it, and create a serializable handle for it</p>
+        let serializable_handle: SerializableHandle = parse_result.document.clone().try_into().unwrap();
```

Comment:
Can you avoid a clone here?

Reply:
Yes we were able to omit the clone since we will no longer need access to document or parse_result.  The example we were referencing used these later, but we do not, so the clone was removed in the updated code.

Reply:
Fantastic! Nice.